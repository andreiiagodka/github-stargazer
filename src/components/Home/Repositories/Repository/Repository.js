import React from "react";
import { Link } from "react-router-dom";
import { Row, Col, ListGroup } from "react-bootstrap";

const repository = props => (
  <Link to={'/details/' + props.id}>
    <ListGroup.Item action>
      <Row>
        <Col md={7}>{props.name}</Col>
        <Col md={5} className='text-right'>{props.stars}</Col>
      </Row>
    </ListGroup.Item>
  </Link>
)

export default repository