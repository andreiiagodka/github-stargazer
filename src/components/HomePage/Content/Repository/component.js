import React from 'react'
import PropTypes from 'prop-types'

import { Link } from 'react-router-dom'
import { Row, Col, ListGroup } from 'react-bootstrap'

const RepositoryComponent = ({ id, name, stars }) => (
  <Link to={'/details/' + id}>
    <ListGroup.Item action>
      <Row>
        <Col md={7}>{name}</Col>
        <Col md={5} className='text-right'>{stars}</Col>
      </Row>
    </ListGroup.Item>
  </Link>
)

RepositoryComponent.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  stars: PropTypes.number.isRequired
}

export default RepositoryComponent