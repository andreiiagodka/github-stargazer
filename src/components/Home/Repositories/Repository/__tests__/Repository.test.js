import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

import Repository from '../Repository'

describe('<Repository /> component', () => {
  it('renders component correctly', () => {
    const props = {
      id: '1',
      name: 'Name',
      stars: 1
    }
    const tree = renderer.create(
      <MemoryRouter>
        <Repository {...props} />
      </MemoryRouter>
    ).toJSON()
  
    expect(tree).toMatchSnapshot()
  })
})
