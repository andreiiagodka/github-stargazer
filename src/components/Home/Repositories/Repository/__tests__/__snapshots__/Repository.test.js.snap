// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repository /> component renders component correctly 1`] = `
<a
  href="/details/1"
  onClick={[Function]}
>
  <button
    className="list-group-item list-group-item-action"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="row"
    >
      <div
        className="col-md-7"
      >
        Name
      </div>
      <div
        className="text-right col-md-5"
      >
        1
      </div>
    </div>
  </button>
</a>
`;
